### DISCUSSION TITLE
Different Unit Systems for Space

### DISCUSSION THREAD

**Discussion by VVishion** [bevyengine/bevy#2794] [Ideas]
Having a naive and unexperienced view on rendering: Is it feasable and does it make sense to implement different __unit systems__ for space?

I think a unit could be "applied" in the projection computation. But I doubt its that easy...

unit systems such as:
* Pixel (for 2D),
* Perceived Pixel (I dont know if metadata such as actual physical display size can be known, 2D),
* Standard (what the renderer works with and what other unit systems are mapped to; unfortunatelly I have no idea what this unit is referred to, 3D, 2D, ...),
* custom Units
* ...,

It would be more expressive than what is currently used.

**Comment by alice-i-cecile** [comment]
#1680, and the topic came up in the [primitive shapes RFC]() is closely related :) I'm personally a fan of this style of  API: it makes code easier to read and prevents frustrating bugs.

In addition to units of measurement, encoding frame of reference and vector vs coordinate would also be nice.

The big questions here are around what exactly that would look like, and if we can manage it without seriously compromising performance, compile time or API quality.
